web-server/src/controllers/users/edit.ts

```diff
@@ -27,12 +28,11 @@ export const edit = async (req: Request, res: Response, next: NextFunction) => {
     if (username) {
       try {
         user.setUsername(username);
-      } catch (err: unknown) {
-        if (err instanceof CustomError) {
-          errors.extend(err.JSON.errors);
+      } catch (e) {
+        if ('username' in e) {
+          err.username = e.username;
         } else {
-          const customError = new CustomError(400, 'Raw', 'Error', err, errors);
-          return next(customError);
+          err.raw = e;
```

Comment:
Again, try not to expose raw stack traces from our server. We have access to these with proper logging tools. External users should not, otherwise they might be able to use it to hack our servers.